src/ClientData/CaptureData.cpp

```diff
@@ -270,6 +270,17 @@ const std::vector<uint64_t>* CaptureData::GetSortedTimerDurationsForScopeId(
   return all_scopes_->GetSortedTimerDurationsForScopeId(scope_id);
 }
 
+std::shared_ptr<ScopeStatsCollection> CaptureData::GetAllScopeStatsCollection() const {
+  return all_scopes_;
+}
+
+std::shared_ptr<ScopeStatsCollection> CaptureData::CreateScopeStatsCollection(
```

Comment:
Same here? Could this return a value?

```cpp
[[nodiscard]] ScopeStatsCollection CaptureData::CreateScopeStatsCollection() const;
```

Reply:
done

Reply:
yep my last commit has it changed to a unique_ptr rather than by value